features/vaadin-dashboard/src/main/java/org/opennms/features/vaadin/dashboard/config/ui/editors/CriteriaBuilderHelper.java

```diff
@@ -128,15 +132,44 @@ public OnmsSeverity getDefault() {
         });
 
         setCriteriaParser(Date.class, new CriteriaParser<Date>() {
+            final DateTimeFormatter parserFormatter = new DateTimeFormatterBuilder()
+                    .append(DateTimeFormatter.ISO_OFFSET_DATE_TIME)
+                    .parseDefaulting(ChronoField.HOUR_OF_DAY, 0)
+                    .parseDefaulting(ChronoField.MINUTE_OF_HOUR, 0)
+                    .toFormatter();
+
+            final String DELTA_PATTERN = "^[+-]\\d+";
```

Comment:
Maybe a good idea to anchor the end of this pattern too?  `^[+-]\\d+$`

Reply:
You are so right.